### PR TITLE
Support Swift 4.1

### PR DESCRIPTION
**Pull request by yigalomer**

### CODE REVIEW

File: Anyway/DetailViewController.swift [hunk 1]
```diff
@@ -49,9 +49,9 @@ class DetailViewController: UIViewController, UITableViewDelegate, UITableViewDa
     func handleMarkerChanged() {
         guard let marker = detailData else {return}
         
-        network.getMarkerDetails(markerId: marker.id) { [weak self] in
-            self?.persons = $0.0
-            self?.vehicles = $0.1
+        network.getMarkerDetails(markerId: marker.id) { [weak self] personList,vehicleList in
+            self?.persons = personList ;//$0.0
+            self?.vehicles = vehicleList ;//$0.1
```
Comment (avielg) [inline Anyway/DetailViewController.swift:54]: nit: drop the comments

File: Anyway/MarkerViews.swift [hunk 2]
```diff
@@ -164,9 +164,13 @@ class MarkerView: MKAnnotationView {
         self.init(annotation: marker, reuseIdentifier: reuseIdentifier)
         
         isEnabled = true
+        //rightCalloutAccessoryView = UIButton(type: .detailDisclosure) as UIView
+        
+        rightCalloutAccessoryView = UIButton.init(type: UIButtonType.detailDisclosure)
+
```
Comment (avielg) [inline Anyway/MarkerViews.swift:169]: nit: kill whitespace & commented line

File: Anyway/Utilities.swift [hunk 3]
```diff
@@ -403,11 +403,11 @@ extension String {
         
         return numOfMatches > 0
     }
-    
-    func attributedStringFromHTMLString(_ overrideFont: Bool = false, color: UIColor = UIColor.white) -> NSAttributedString {
+
+   /* func attributedStringFromHTMLString(_ overrideFont: Bool = false, color: UIColor = UIColor.white) -> NSAttributedString {
```
Comment (avielg) [inline Anyway/Utilities.swift]: why function commented?

File: Anyway/ViewController+MapView.swift [hunk 4]
```diff
@@ -15,6 +15,7 @@ import Foundation
 */
 extension ViewController: MKMapViewDelegate {
     
+
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift]: nit: kill

File: Anyway/ViewController+MapView.swift [hunk 5]
```diff
@@ -84,8 +86,7 @@ extension ViewController: MKMapViewDelegate {
                 return mView
             }
             return MarkerView(marker: marker)
-            
-        }
+         }
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift]: nit: indent

File: Anyway/ViewController+MapView.swift [hunk 6]
```diff
@@ -43,6 +44,7 @@ extension ViewController: MKMapViewDelegate {
         let distance = CLLocation.distance(from: lastRegion.center, to: mapView.region.center)
         print("distance: \(distance)")
         
+        //?? YIGAL what is this ?
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift]: nit: drop

File: Anyway/ViewController+MapView.swift [hunk 7]
```diff
@@ -105,6 +106,7 @@ extension ViewController: MKMapViewDelegate {
      Opens the screen to show more details about the accident.
      */
     func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift:109]: nit: whitespace

File: Anyway/ViewController+MapView.swift [hunk 8]
```diff
@@ -126,6 +128,7 @@ extension ViewController: MKMapViewDelegate {
         }
     }
     
+
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift]: nit: whitespace

File: Anyway/ViewController+MapView.swift [hunk 9]
```diff
@@ -138,6 +141,7 @@ extension ViewController: MKMapViewDelegate {
      */
     func mapView(_ mapView: MKMapView, didSelect view: MKAnnotationView) {
         // handle a marker group
+        print("mapView:didSelect")
```
Comment (avielg) [inline Anyway/ViewController+MapView.swift]: nit: drop comment

File: Anyway/ViewController.swift [hunk 10]
```diff
@@ -75,6 +75,8 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
     override func viewDidLoad() {
         super.viewDidLoad()
         
+         print("viewDidLoad")
+
```
Comment (avielg) [inline Anyway/ViewController.swift]: nit: whitespace & comment

**Review by avielg — CHANGES_REQUESTED** [review]
few nits, otherwise looks good

Thanks!

**Review by avielg — APPROVED** [review]
where is -180,-180? 🤔

**Review by avielg — CHANGES_REQUESTED** [review]
goddammit github…. I didn't realize the 2 commits are together… anyway the 2 one looks good, just go over the nits in the first one

File: Anyway/Utilities.swift (hunk 3 above)
Comment (yigalomer) [reply Anyway/Utilities.swift]: Never used, made some compilation errors

Comment (avielg) [reply Anyway/Utilities.swift]: oh then feel free to delete it 👍 
if we ever need it again we have source control (:

**Comment by yigalomer** [comment]
I saw that member lastRegion =-180,-180  when app start  when we call "var lastRegion = MKCoordinateRegionForMapRect(MKMapRectNull)"

I needed a way to get annotations when app starts

Didn't find another way to check for null on lastRegion - do you know a way?

File: Anyway/ViewController.swift (hunk 10 above)
Comment (yigalomer) [reply Anyway/ViewController.swift]: what is "nit"?

File: Anyway/ViewController+MapView.swift (hunk 5 above)
Comment (yigalomer) [reply Anyway/ViewController+MapView.swift]: Fixed

File: Anyway/ViewController+MapView.swift (hunk 4 above)
Comment (yigalomer) [reply Anyway/ViewController+MapView.swift]: fixed

File: Anyway/Utilities.swift (hunk 3 above)
Comment (yigalomer) [reply Anyway/Utilities.swift]: fixed

File: Anyway/MarkerViews.swift (hunk 2 above)
Comment (yigalomer) [reply Anyway/MarkerViews.swift:169]: fixed

File: Anyway/ViewController+MapView.swift (hunk 7 above)
Comment (yigalomer) [reply Anyway/ViewController+MapView.swift:109]: fixed

File: Anyway/ViewController+MapView.swift (hunk 8 above)
Comment (yigalomer) [reply Anyway/ViewController+MapView.swift]: fixed

File: Anyway/ViewController+MapView.swift (hunk 9 above)
Comment (yigalomer) [reply Anyway/ViewController+MapView.swift]: fixed

File: Anyway/ViewController.swift (hunk 10 above)
Comment (yigalomer) [reply Anyway/ViewController.swift]: fixed

**Review by avielg — APPROVED** [review]

**Pull request merged**